module2/persistence/src/main/java/com/epam/esm/dao/GiftCertificateDao.java

```diff
@@ -0,0 +1,29 @@
+package com.epam.esm.dao;
+
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.entity.Tag;
+import org.springframework.lang.Nullable;
+
+import java.util.List;
+import java.util.Set;
+
+public interface GiftCertificateDao {
+
+    long save(GiftCertificate certificate, Set<Tag> tags);
+
+    List<GiftCertificate> getAll();
```

Comment:
the same methods as in TagDao, maybe it is better to move equal methods to common abstract Dao?